src/shogun/machine/NonParametricMachine.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ */
+
+#ifndef NONPARAMETRCMACHINE_H_
+#define NONPARAMETRCMACHINE_H_
+
+#include <shogun/machine/Machine.h>
+
+namespace shogun
+{
+
+	class NonParametricMachine : public Machine
+	{
+	public:
+		NonParametricMachine()
+		{
+			SG_ADD(
+			    &m_labels, "labels", "labels used in train machine algorithm",
+			    ParameterProperties::READONLY);
+			SG_ADD(
+			    &m_features, "features",
+			    "features used in train machine algorithm",
+			    ParameterProperties::READONLY);
+		}
+		virtual ~NonParametricMachine()
+		{
+		}
+
+	private:
+		std::shared_ptr<Features> m_features;
+
+		// when all refactor is done, we should use this m_labels
+		// std::shared_ptr<Labels> m_labels;
```

Comment:
and then also remove `m_labels` from `Machine`, but I guess we won't forget that as we will get compiler errors otherwise :D